Add database set/query and cap query

yuvalavn opened the pull request.

Commit:
common/mlx5: add mlx5 regex command structs

Add regex commands structs to support regex.

Signed-off-by: Yuval Avnery <yuvalav@mellanox.com>

Commit:
regex/mlx5: add Makefiles

Add Makefiles to the regex tree.

Signed-off-by: Yuval Avnery <yuvalav@mellanox.com>

Commit:
regex/mlx5: add mlx5 regex src file

mlx5_regex.c will abstract API to FW/HW regex operations.

Signed-off-by: Yuval Avnery <yuvalav@mellanox.com>

Commit:
common/mlx5: add support for regex capability query

Update hca cap struct and common query hca cap function.

Signed-off-by: Yuval Avnery <yuvalav@mellanox.com>

Commit:
regex/mlx5: add database commands support

Add regex database commands:
 - Stop engine
 - Set DB umem
 - Resume

Database init flow per engine:
1. Set database umem.
2. Resume.

Database update flow:
1. Stop engine.
1. Set database.
3. Resume.

Signed-off-by: Yuval Avnery <yuvalav@mellanox.com>

paravmellanox commented on drivers/regex/mlx5/mlx5_regex.c:
```diff
@@ -3,3 +3,15 @@
 
 #include "mlx5_regex.h"
+
+static int mlx5_regex_is_supported(struct ibv_context *ibv_ctx)
```
do we have the bool data type in  dpdk?
if so, please return as bool

orikam replied:
There are some cases of bool usage in DPDK but not many
I prefer int so we can use it as error code,

paravmellanox reviewed (commented):
please update the commit log of what each stop, resume apis do and when to use it, limitation etc.

paravmellanox commented on drivers/regex/mlx5/Makefile:
```diff
@@ -0,0 +1,54 @@
+
+EXPORT_MAP := rte_pmd_mlx5_version.map
+
+# memseg walk is not part of stable API
```
can we avoid it?

yuvalavn replied:
I relied on the mlx5 PMD here. assuming we need the same libs.
So maybe forgot to remove some things.

yuvalavn replied:
Yes, it is needed

paravmellanox reviewed (commented):
@orikam 
it is a static function so not so useful on error code.
and error code on false can be easily converted to ENOSUPP

Commit:
regex/mlx5: Add RXP register read/write commands

RXP is configured through a series or registers write/read
operations.
These registers are also used for statistics.

Signed-off-by: Yuval Avnery <yuvalav@mellanox.com>

Commit:
regex/mlx5: Add log utils

Add the DRV_LOG macro which should be used for error prints.

Signed-off-by: Yuval Avnery <yuvalav@mellanox.com>
